### ISSUE TITLE
Nukkit support

### ISSUE THREAD

**Issue by MrPowerGamerBR** [aikar/TaskChain#13]
(I already added Nukkit support to my TaskChain fork https://github.com/PocketDreams/TaskChain however I didn't create a pull request because I don't know if you are accepting pull requests for other server softwares...)

(And my TaskChain fork only works with my Nukkit fork until https://github.com/Nukkit/Nukkit/pull/1459 is pulled to upstream, so...)

**Comment by willies952002** [comment]
Personally, I wouldn't see why a PR for adding support for another platform would be an issue :D

**Comment by aikar** [comment]
If you can do the following, I can pull it if you PR:
 - [ ] Fix the whitespace in root pom.xml
 - [ ] Remove eclipse project stuff
 - [ ] Rename the Bukkit inner class to Nukkit
 - [ ] Provide Nukkit in a public repository that the nukkit pom file includes like Bukkit does so I can compile w/o having Nukkit on my system.

**Comment by MrPowerGamerBR** [comment]
@aikar yeah, my fork was like "well, it works, so..." (and I tried to clean up the Eclipse mess... but it didn't work)

The problem is that the current code **won't** work on the current version of Nukkit, Nukkit's tasks doesn't include the plugin instance on the task so I think some parts of the code won't work (right?), so I will probably do a clean pull request when my pull request @ Nukkit is pulled. (and yes, I tested my modifications on Nukkit... it works fairly well (I already use TaskChain on my PC server), waaay better than creating a bunch of schedulers)

**Comment by MrPowerGamerBR** [comment]
The PR was pulled to Nukkit, so I will probably create a PR to add Nukkit to TaskChain soon.

The problem is the Maven repo, there isn't an official Maven repo for Nukkit, so that will be a bit difficult.

**Comment by aikar** [comment]
How do people currently build plugins then? do you really make them compile locally just to get artifacts?

**Comment by MrPowerGamerBR** [comment]
@aikar by "you" you mean "they", right? I'm not a Nukkit developer.

But yeah, most of them installs the Nukkit artifact in the local repo or use unofficial maven repository. (Mengcraft's Maven repo)

**Comment by aikar** [comment]
long as it compiles then unofficial should be fine? if thats what the community is use to using.

**Comment by aikar** [comment]
@MrPowerGamerBR  @imjack

You both now have Nukkit versions. I've added Mengcraft to my Nexus Repo as a proxy, so if one of you can PR it (without the mengcraft reference, since it will be in my repo cache instead), we can officially get this support added.

**Comment by MrPowerGamerBR** [comment]
@aikar I stopped using Nukkit since February because it is on "life support" (it is only receiving protocol updates *sometimes*) so I migrated to other platforms, sorry 😢

**Comment by aikar** [comment]
@MrPowerGamerBR seems active again? http://ci.mengcraft.com:8080/job/nukkit/changes

**Comment by MrPowerGamerBR** [comment]
@aikar not really "active" when you compare PocketMine-MP (which Nukkit was based of)

Nukkit only receives some protocol fixes (when a new MCPE version is released) and some very critical fixes, but everyone of the main Nukkit team is inactive.
